test/inductor/extension_backends/extension_scheduling.py

```diff
@@ -0,0 +1,37 @@
+from torch._inductor.codegen import cpp, wrapper
+from torch._inductor.codegen.scheduling import Scheduling
+from torch._inductor.virtualized import V
+
+
+class ExtensionWrapperCodegen(wrapper.WrapperCodeGen):
+    def __init__(self):
+        super().__init__()
+
+
+class ExtensionScheduling(Scheduling):
+    def __init__(self, scheduler):
+        self.scheduler = scheduler
+        self._scheduling = cpp.CppScheduling(scheduler)
+
+    def can_fuse_vertical(self, node1, node2):
+        return True
+
+    def can_fuse_horizontal(self, node1, node2):
+        return True
+
+    def group_fn(self, *args, **kwargs):
+        sizes = args[0]
+        return tuple(tuple(map(V.graph.sizevars.simplify, s)) for s in sizes)
+
+    def codegen_template(self, *args, **kwargs):
+        pass
+
+    def codegen_nodes(self, *args, **kwargs):
+        nodes = args[0]
+        self._scheduling.codegen_nodes(nodes)
+
+    def codegen_sync(self, *args, **kwargs):
+        pass
+
+    def flush(self, *args, **kwargs):
+        self._scheduling.flush()
```

Comment:
I was hoping to see a more complete example than this.  One that actually generates some minimal working C++ code.  I worry there are hidden issues that will be uncovered.

Reply:
Do you think partial elementwise operations are good enough? Or should we implement all the functions?

Reply:
It doesn't need to implement everything, but it should demonstrate the key concepts:
1) pointwise ops
2) reduction ops
3) horizontal fusion
4) vertical fusion

Reply:
Okay. I will develop a minimal codegen to demonstrate these key concepts.